pkg/services/dashboards/database/database.go

```diff
@@ -589,10 +589,12 @@ func (d *dashboardStore) deleteDashboard(cmd *dashboards.DeleteDashboardCommand,
 			return err
 		}
 
-		// remove all access control permission with folder scope
-		err := d.deleteResourcePermissions(sess, dashboard.OrgID, dashboards.ScopeFoldersProvider.GetResourceScopeUID(dashboard.UID))
-		if err != nil {
-			return err
+		if !cmd.SkipRemovePermissions {
+			// remove all access control permission with folder scope
+			err := d.deleteResourcePermissions(sess, dashboard.OrgID, dashboards.ScopeFoldersProvider.GetResourceScopeUID(dashboard.UID))
```

Comment:
since we moved the permissions removal to the folderStorage... maybe we could remove this block entirely